src/app/ReadClient.h

```diff
@@ -59,17 +60,22 @@ class ReadClient : public Messaging::ExchangeDelegate
     {
     public:
         virtual ~Callback() = default;
+
         /**
-         * Notification that a list of events is received on the given read client.
          * The ReadClient object MUST continue to exist after this call is completed.
          *
-         * @param[in]  apReadClient         The read client which initialized the read transaction.
-         * @param[in]  aEventReports        TLV reader positioned at the list that contains the events.  The
-         *                                  implementation of EventStreamReceived is expected to call Next() on the reader to
-         *                                  advance it to the first element of the list, then process the elements from beginning to
-         *                                  the end. The callee is expected to consume all events.
+         * This callback will be called when receiving event data received in the Read and Subscribe interactions
+         *
+         * @param[in] apReadClient: The read client object that initiated the read or subscribe transaction.
+         * @param[in] aEventHeader: The event header in report response.
+         * @param[in] apData: A TLVReader positioned right on the payload of the event. This will be set to null if the apStatus is
+         * not null.
```

Comment:
Can they both be null?  This should probably clearly document that one and only one of the two will be non-null...